### ISSUE TITLE
Beta phase tag colours don't meet WCAG 2.0 AA

### ISSUE THREAD

**Issue by craigmorrison** [alphagov/govuk_elements#217]
The white text on orange (#f47738) background used in the beta phase tag doesn't meet the contrast criteria for WCAG AA.

It give us a contrast ratio of 2.8:1

WCAG AA requires a ratio of 4.5:1
WCAG AAA requires a ratio of 7:1

**Comment by sanjaypoyzer** [comment]
Yeah, this is a similar issue to the confirmation boxes in #200, in which we're discussing increasing the font size.

If we did similar for this, we could have something like: 

<img width="438" alt="screen shot 2016-05-18 at 14 55 05" src="https://cloud.githubusercontent.com/assets/4106955/15361251/a2f95452-1d08-11e6-82d3-6cbb13892108.png">

(Obviously we'd also increase the size of `.alpha-tag` but I've included it unchanged in the screenshot for comparison.)

As with #200, it's either this or pick a duller shade of orange I reckon.

Would be good to get @markhurrell's view here too.

**Comment by markhurrell** [comment]
hey, I'm not going to be around for a few days. @stephenjoe1 @alextea what do you think?

**Comment by edwardhorsford** [comment]
Increased size looks ok to me.

Question: do we consider this element decorative? The content makes sense without the label. If it's decorative, it's less crucial to meet contrast guidelines.

**Comment by markhurrell** [comment]
I mean, I sort of do consider it decorative - it is sort of jargon




Mark Hurrell
Head of Design for GOV.UK
Government Digital Service

**Comment by gemmaleigh** [comment]
This issue comes up a lot.

The options we've considered so far are:

1. Change the colour, but we'd be keen to keep the Alpha and Beta colours as they are distinctive and bright
2. Increase the size of the text, so the contrast ratio requirements are lower.

For 2, Even at a larger text size, this background colour fails.

```
Foreground: #FFFFFF - Background: #F47738

The contrast ratio is: 2.78:1

Text failed at level AA
Text failed at level AAA
Large text failed at level AA
Large text failed at level AAA

1.4.3 Contrast (Minimum):  Text (and images of text) have a contrast ratio of at least 4.5:1, except if the text is pure decoration.  Larger scale text (at least 18 point or 14 point bold) or images of text can have a contrast ratio of 3:1. (Level AA)

1.4.6 Contrast (Enhanced): Text (and images of text) have a contrast ratio of at least 7:1, except if the text is pure decoration.  Larger scale text (at least 18 point or 14 point bold) or images of text can have a contrast ratio of 4.5:1. (Level AAA)

Note: Fonts that are extraordinarily thin or decorative are harder to read at lower contrast levels.
```

I'd be keen for @aduggin or Richard Morton to let us know what might be a better way to resolve this.

**Comment by edwardhorsford** [comment]
Two more options to consider:
* Remove them entirely. Users don't know what they mean. Why are they there?
* Do nothing. They're largely decorative anyway. Are they delivering important content to users? no.

**Comment by sanjaypoyzer** [comment]
I think *some* users know what they mean. Definitely not all, but it's worth bearing in mind that some do, and those users will find it useful to know what stage of development a service is in. Definitely doesn't seem like the most important issue in the world though, but it would be great to hear from some real users.

**Comment by accessiblewebuk** [comment]
It isn't decorative if it contains information, which could the word(s) or could the meaning being conveyed. For a sighted user there is meaning in the banner (whether or not the meaning is understood is a slightly different matter), so to be accessible we mustn't discriminate between users just because some can't perceive it through sight. 

The question about whether to remove it altogether - there is a danger there of services remaining in beta with less of an incentive to move to live. Personally I think it is important to know if a service is beta (or perhaps "trial service" might be more easily understood) - as Caroline says, whatever the question, the answer is always more user research. 

Regarding colours - the easy answer is to use dark text with orange. Very dark grey works - but of course still potentially has problems for anyone reversing the colours. #0b0b0c against #ff7738 has contrast of 7.0:1 which passes at level AA and level AAA at all text sizes.

**Comment by accessiblewebuk** [comment]
missed out "be" twice in my previous comment about conveying meaning - oh the irony!

**Comment by accessiblewebuk** [comment]
@gemmaleigh I raised the issue for the hack day.

**Comment by aduggin** [comment]
This isn't decorative - it's communicating information so it does need to meet colour contrast requirements. 

Fixing it will benefit people with low vision. It will also mean that when we are running automated accessibility tools as part of continuous integration we won't have every page with the beta banner failing.

**Comment by craigmorrison** [comment]
Regarding how well understood 'beta' is, our user research for Student Finance has also shown that it varies quite a bit. Some people have no understanding and others roughly understand beta to mean that it's a trial or it's in testing.

The concern that has been raised within SLC with beta application forms is that when the user does understand what beta means, users aren't clear on whether it means the financial product itself is in beta, or if it's just the application service that's in beta. For us it's important that it's understood to be the latter.

**Comment by accessiblewebuk** [comment]
Perhaps "Beta" should link to a page giving more information for those who are interested - on the other hand that might confuse? 

Interestingly I just noticed that https://www.gov.uk/vehicle-tax doesn't have a "Beta" banner, but clicking on the start button links to https://www.vehicletax.service.gov.uk/ which does. Is that the normal way for services to do this, i.e. is it not the policy to show it on the initial landing page for the service?

**Comment by craigmorrison** [comment]
I think a link would be okay but I wouldn't expect it to be used much. We don't see much use of the footer links within our service.

I think the main question over this for me is, what does "beta" add to "This is a new service – [your feedback](#) will help us to improve it". The sentence conveys the same thing in a clearer manner.

**Comment by gemmaleigh** [comment]
To meet colour contrast, the colour was updated in the [govuk_frontend_toolkit v5.0.2](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/CHANGELOG.md#502).

The background colour for alpha and beta tags is now `$govuk-blue`.
